src/main/java/com/github/splendor_mobile_game/websocket/handlers/reactions/MakeReservationFromDeck.java

```diff
@@ -76,9 +89,178 @@ public MakeReservationFromDeck(int connectionHashCode, UserMessage userMessage,
+            
+            ResponseData responseData = new ResponseData(
+                reservee.getUuid(), 
+                new CardDataResponse(
+                    card.getUuid(), 
+                    card.getPoints(), 
+                    card.getAdditionalToken(), 
+                    new TokensDataResponse(
+                        card.getCost(TokenType.RUBY), 
+                        card.getCost(TokenType.EMERALD), 
+                        card.getCost(TokenType.SAPPHIRE), 
+                        card.getCost(TokenType.DIAMOND), 
+                        card.getCost(TokenType.ONYX)
+                    )
+                )
+            );
+
+            ArrayList<User> players = room.getAllUsers();
+            ServerMessage serverMessage = new ServerMessage(
+                userMessage.getContextId(), 
+                ServerMessageType.MAKE_RESERVATION_FROM_DECK_ANNOUNCEMENT, 
+                Result.OK, 
+                responseData
+            );
+
+            for(User player : players){
+                messenger.addMessageToSend(player.getConnectionHashCode(), serverMessage);        
+            }
+
+        }catch (Exception e) {
+            ErrorResponse errorResponse = new ErrorResponse(Result.FAILURE, e.getMessage(), ServerMessageType.MAKE_RESERVATION_FROM_DECK_RESPONSE, userMessage.getContextId().toString());
+            messenger.addMessageToSend(connectionHashCode, errorResponse);
+        }
+    }
+
+    private void validateData(DataDTO dataDTO, Database database) throws Exception{
+        User user = database.getUser(dataDTO.userUuid);
+
+        //Check if user exists
+        if(user==null){
+            throw new UserDoesntExistException("There is no such user in the database");
+        }
+
+        Room room = database.getRoomWithUser(user.getUuid());
+
+        //Check if user is in any room
+        if(room==null){
+            throw new RoomDoesntExistException("User does not belong to any room");
+        }
+
+        Game game = room.getGame();
+        
+        //check if the room is in the game
+        if(game==null){
+             throw new RoomInGameException("Room is not in a game state");
+        }
+
+        //Check if it is user's turn
+        if(game.getCurrentPlayer()!=user){
+             throw new UserTurnException("It is not user's turn");
+        }
+        
+        //Check reservation count
+        if(user.getReservationCount()>=3){
+            throw new UserReservationException("User has to many reserved cards in hand");
+        }
+
+        //Check throughout game reservation count
+        if(user.getThroughoutGameReservationCount()>=5){
+            throw new UserReservationException("User reached the limit of reserved cards per game");
+        }
+
+        //Check if user has not maxed tokens
+        if(user.getTokenCount()>=10){
+            throw new TokenCountException("Reached max token count on hand");
```

Comment:
I don't remeber exact rules, but I think user can reserve a card in such situation, they simply just don't get the gold token. 

Do you remember what should be done in such case? @Vertonowsky @KamilDevelopments @EaZZyHK @sssadpeach15